cmake/projects/Boost/schemes/url_sha1_boost_ios_library.cmake.in

```diff
@@ -283,6 +283,38 @@ if(has_isim)
       install
       "--prefix=@HUNTER_PACKAGE_INSTALL_PREFIX@"  
   )
+elseif(has_isim)
```

Comment:
whats the difference between the `has_isim AND has_iphoneos` and the `has_isim` External Project call? If it is just one variable or so we could use a variable instead of code duplication

Reply:
Previously this scheme assumed that the only options were `(has_iphoneos && has_isim)` or `(has_iphoneos && !has_isim)`, so iOS device only and iOS device/simulator togehter. Now we are making sure that iOS simulator only works (due to the arm64 thing mentioned in the issue).

The difference is that now we have added the `URL` and `URL_HASH` into the simulator version and we do not set `DOWNLOAD_COMMAND ""`. Before this change the source was asssumed to have been downloaded by the `has_iphoneos` stage, and so wasn't available.

I tried to do a variable

```cmake
if (has_iphoneos)
  set(skip_download "DOWNLOAD_COMMAND" "")
endif()

ExternalProject_Add(
     "@HUNTER_EP_NAME@-ios_sim"
    ${skip_download}
...
```

But I couldn't get it to work, something about the empty string in the array being ignored I think (if I set it to nonsense it correctly failed). It would be nicer in a variable. Someone with better knowledge of how CMake treats these things might be able to make that work though.

Reply:
now I understand, I tried many variations to set the skip_download variable, and I didn't find a working one

```cmake

set(skip_download "DOWNLOAD_COMMAND")
set(skip_download "DOWNLOAD_COMMAND;")
set(skip_download "DOWNLOAD_COMMAND" "")
set(skip_download "DOWNLOAD_COMMAND;\"\"")
set(skip_download "DOWNLOAD_COMMAND" "\"\"")
message(WARNING "${skip_download}")
include(ExternalProject)
ExternalProject_Add(
  foo
  "${skip_download}"
  #DOWNLOAD_COMMAND ""
  DOWNLOAD_DIR "${CMAKE_SOURCE_DIR}/foo/download"
  SOURCE_DIR "${CMAKE_SOURCE_DIR}/foo/src"
  SOURCE_DIR "${CMAKE_SOURCE_DIR}/foo/install"
)
```

Reply:
Found one solution, just add a command which tells us we skip the download step

```cmake
set(skip_download "DOWNLOAD_COMMAND" "${CMAKE_COMMAND}" -E echo "SKIPPING DOWNLOAD STEP")
```

Would this be acceptable?

Reply:
Yeah I prefer that, I can rework it on Monday

Reply:
I have reworked it, and it appears to work, thanks!

However, I have also updated my Xcode and it has caused another issue with my iOS build... the mission continues :(